### PR TITLE
@redyaffle please look this over. I'm going to add tests before submitting it for merge

### PR DESCRIPTION
**Pull request by bantic**
_do not merge_

Token-based Authentication
- Adds torii, uses torii's session as `toriiSession`
- Adds torii provider and adapter for lion-guardians
- Adds a 'Authorization' token header to the adapter
- stores the provided token in local storage

### CODE REVIEW

**Comment by bantic** [comment]
needs these changes from the backend: https://github.com/201-created/LG-api/pull/6

**Comment by bantic** [comment]
@redyaffle This is ready to be merged now.

File: app/adapters/application.js [hunk 1]
```diff
@@ -1,6 +1,19 @@
...
-  host: config.apiURL
+  host: config.apiHost,
+
+  headers: function(){
+    let headers = {};
+    let token = storage.read(config.authTokenKey);
+    if (token) {
+      // For rails token authorization
+      headers['Authorization'] =
+        `Token token=${token.token}, email=${token.email}`;
```
Comment (bantic) [inline app/adapters/application.js:14]: this is an [ES6 template string](http://updates.html5rocks.com/2015/01/ES6-Template-Strings). We can write ES6 code in ember-cli now due to the 6to5 plugin.

**Comment by redyaffle** [comment]
@bantic Looks great - merging now!

**Pull request merged**